core/src/Revolution/modX.php

```diff
@@ -2746,25 +2746,14 @@ protected function _initSession($options = null) {
         $contextKey= $this->context instanceof modContext ? $this->context->get('key') : null;
         if ($this->getOption('session_enabled', $options, true) || isset($_GET['preview'])) {
             if (!in_array($this->getSessionState(), [modX::SESSION_STATE_INITIALIZED, modX::SESSION_STATE_EXTERNAL, modX::SESSION_STATE_UNAVAILABLE], true)) {
-                $sh = false;
-                if ($sessionHandlerClass = $this->getOption('session_handler_class', $options)) {
-                    if ($shClass = $this->loadClass($sessionHandlerClass, '', false, true)) {
-                        if ($sh = new $shClass($this)) {
-                            session_set_save_handler(
-                                [& $sh, 'open'],
-                                [& $sh, 'close'],
-                                [& $sh, 'read'],
-                                [& $sh, 'write'],
-                                [& $sh, 'destroy'],
-                                [& $sh, 'gc']
-                            );
-                        }
+                $sessionHandlerClass = $this->getOption('session_handler_class', $options);
+                if (is_string($sessionHandlerClass) && !empty($sessionHandlerClass) && class_exists($sessionHandlerClass)) {
+                    $ch = new $sessionHandlerClass($this);
+                    if ($ch instanceof \SessionHandlerInterface) {
+                        $this->services->add('session_handler', $ch);
                     }
                 }
-                if (
-                    (is_string($sessionHandlerClass) && !$sh instanceof $sessionHandlerClass) ||
-                    !is_string($sessionHandlerClass)
-                ) {
+                if (!$this->services->has('session_handler')) {
```

Comment:
I'm not sure this is correct. You do not need a custom session handler to be able to set the session_save_path. In fact, this is meant to be used with the default PHP session handling to customize the location of the session files.

Reply:
can you show me an example to make this condition true
```
if (
    (is_string($sessionHandlerClass) && !$sh instanceof $sessionHandlerClass) ||
     !is_string($sessionHandlerClass)
)
```

Reply:
If it is not set to a modSessionHandler class or it is empty so it uses what is defined in PHP's ini.

Reply:
1. There is a deprecated loadClass method.
2. `$sh` object will always be the instance of `$sessionHandlerClass`, so this condition will be true only if `$sessionHandlerClass` in not a string.
3. This condition I move to line 2750.